src/main/java/com/exasol/glue/ExasolJdbcUrlParser.java

```diff
@@ -0,0 +1,58 @@
+package com.exasol.glue;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.exasol.errorreporting.ExaError;
+
+/**
+ * A helper class to extract Exasol {@code JDBC} URL components.
+ */
+public final class ExasolJdbcUrlParser {
+    private static final String EXASOL_JDBC_URL_REGEX = "jdbc:exa:(?<host>[^:\\/;]+)(\\/(?<fingerprint>[^:;\\/]*))?:(?<port>\\d+)(;(?<options>.*))?";
+    private static final Pattern EXASOL_JDBC_URL_PATTERN = Pattern.compile(EXASOL_JDBC_URL_REGEX);
+
+    /**
+     * Extracts {@code HOST}, {@code PORT} and other components from Exasol {code @JDBC} connection string.
+     *
+     * @param jdbcUrl Exasol JDBC connection string
+     * @return map of components
+     */
+    public Map<String, String> parse(final String jdbcUrl) {
+        final Matcher matcher = EXASOL_JDBC_URL_PATTERN.matcher(jdbcUrl);
+        if (!matcher.matches()) {
+            throw new IllegalArgumentException(ExaError.messageBuilder("E-EGC-29")
+                    .message("Unable to extract JDBC URL '{{jdbcUrl}}' components.", jdbcUrl)
+                    .mitigation("Please check that Exasol JDBC URL is constructed properly.").toString());
+        }
+        final Map<String, String> result = new HashMap<>();
+        final String host = matcher.group("host");
+        final String port = matcher.group("port");
+        final String fingerprint = getOptionalGroup(matcher, "fingerprint");
```

Comment:
It should work I think. The next group, `options` would catch it, and it will be added later to the url generation.

But I will add test to check it.

Reply:
Ahhh, please forget my comment. I thought the fingerprint as option would be then ignored. But it will be forwarded to the DB anyway.